pkg/vfs/couchdb_indexer.go

```diff
@@ -343,3 +343,15 @@ func (c *couchdbIndexer) DirLength(doc *DirDoc) (int, error) {
 	}
 	return int(f64), nil
 }
+
+func (c *couchdbIndexer) DirChildExists(dirID, name string) (bool, error) {
+	var res couchdb.ViewResponse
+
+	// consts.FilesByParentView keys are [parentID, type, name]
+	err := couchdb.ExecView(c.db, consts.FilesByParentView, &couchdb.ViewRequest{
+		Key:         []string{dirID, consts.FileType, name},
```

Comment:
This wont match a child folder with the given name, is that on purpose ? You can use  a Keys to match both.